### PR TITLE
Fix discovering numa distance when node ids are not starting from 0 or it's ids are not sequential

### LINKED ISSUE #113781: Latest k8s fails with kublet panic runtime error
### What happened?

While setting up k8s cluster using the ci/latest version, we notice the kublet failure with below error message in its log:

```
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: I1109 07:20:47.984943   26715 topology_manager.go:134] "Creating topology manager with policy per scope" topologyPolicyName="none" topologyScopeName="container"
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: panic: runtime error: index out of range [1] with length 1
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: goroutine 1 [running]:
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/pkg/kubelet/cm/topologymanager.newNUMAInfo({0xc0008ca540, 0x1, 0x0?}, 0xc000eda120)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         pkg/kubelet/cm/topologymanager/numa_info.go:62 +0x2d4
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/pkg/kubelet/cm/topologymanager.NewNUMAInfo(...)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         pkg/kubelet/cm/topologymanager/numa_info.go:45
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/pkg/kubelet/cm/topologymanager.NewManager({0xc0008ca540, 0x1, 0x1}, {0x143fa6eb, 0x4}, {0x14403c29, 0x9}, 0xc0007dd800?)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         pkg/kubelet/cm/topologymanager/topology_manager.go:136 +0x1c8
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/pkg/kubelet/cm.NewContainerManager({_, _}, {_, _}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         pkg/kubelet/cm/container_manager_linux.go:288 +0xba4
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/cmd/kubelet/app.run({0x14c4b570, 0xc0004b03c0}, 0xc000ddc000, 0xc0008cea80, {0x7fff5c446958, 0xc0004b01c0})
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         cmd/kubelet/app/server.go:723 +0x1724
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/cmd/kubelet/app.Run({0x14c4b570, 0xc0004b03c0}, 0xc000ddc000, 0x6?, {0x7fff5c446958, 0xc0004b01c0})
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         cmd/kubelet/app/server.go:419 +0x344
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/kubernetes/cmd/kubelet/app.NewKubeletCommand.func1(0x101c2b8c?, {0xc000192080, 0x6, 0x6})
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         cmd/kubelet/app/server.go:265 +0x9c0
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: github.com/spf13/cobra.(*Command).execute(0xc0005a6c00, {0xc000192080, 0x6, 0x6})
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         vendor/github.com/spf13/cobra/command.go:916 +0x674
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: github.com/spf13/cobra.(*Command).ExecuteC(0xc0005a6c00)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         vendor/github.com/spf13/cobra/command.go:1040 +0x36c
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: github.com/spf13/cobra.(*Command).Execute(...)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         vendor/github.com/spf13/cobra/command.go:968
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/component-base/cli.run(0xc0005a6c00)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         vendor/k8s.io/component-base/cli/run.go:146 +0x2dc
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: k8s.io/component-base/cli.Run(0xc000180058?)
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         vendor/k8s.io/component-base/cli/run.go:46 +0x30
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]: main.main()
Nov 09 07:20:47 config1-1667977224-worker-0 kubelet[26715]:         cmd/kubelet/kubelet.go:36 +0x28
Nov 09 07:20:48 config1-1667977224-worker-0 systemd[1]: kubelet.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Nov 09 07:20:48 config1-1667977224-worker-0 systemd[1]: kubelet.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
```

### What did you expect to happen?

There should be no error while kubelet is coming up.

### How can we reproduce it (as minimally and precisely as possible)?

Install the prerequisites needed for the cluster setup.
Enable and try to start the kubelet.

### Kubernetes version
<details>

```console
$ kubectl version
"v1.26.0-alpha.3.456+20a9f7786aa4ee"
```
</details>

### Cloud provider

<details>
We noticed this in our IBM Cloud provider PowerVS
</details>

### PR DESCRIPTION
**Pull request by PiotrProkop**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #113781

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/topologymanager/numa_info.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func NewNUMAInfo(topology []cadvisorapi.Node) (*NUMAInfo, error) {
 
 func newNUMAInfo(topology []cadvisorapi.Node, sysFs *NUMASysFs) (*NUMAInfo, error) {
 	var numaNodes []int
-	distances := make([][]uint64, len(topology))
+	distances := map[int][]uint64{}
```
Comment (klueska) [inline pkg/kubelet/cm/topologymanager/numa_info.go:50]: Ugh, I should have caught this in the initial review

Comment (PiotrProkop) [reply pkg/kubelet/cm/topologymanager/numa_info.go:50]: I thought NUMA nodes are always ordered from 0 but it seems that is not true 😞

**Comment by klueska** [comment]
/cc @kubernetes/milestone-maintainers
Can you please add this critical bug fix to the v1.26 milestone

**Comment by saschagrunert** [comment]
@klueska please refer to the sig node folks of @kubernetes/milestone-maintainers to accept such a change.

**Comment by klueska** [comment]
/cc dchen1107 derekwaynecarr
This bug was introduced as part of https://github.com/kubernetes/kubernetes/pull/112914

It will cause the `kubelet` to crash on any system that has more than 1 NUMA node (i.e. almost any machine nowadays)

**Comment by PiotrProkop** [comment]
Just to clarify it will fail if NUMA nodes ids are not  sequential e.g `node0`, `node1`  starting from 0 or if the single NUMA node on the system is not `node0`.

**Comment by liggitt** [comment]
I'll defer to node reviewers on this specific change, but I labeled the bug as a regression/release-blocker and added it to the milestone

**Comment by klueska** [comment]
Thanks @liggitt is that label / milestone not visible on this PR itself?

**Comment by liggitt** [comment]
> Thanks @liggitt is that label / milestone not applicable to this PR itself? Or just the bug?

if this is accepted as the fix, it will be added to this PR (that's what I'm deferring to node folks on)

**Comment by dims** [comment]
thanks @PiotrProkop

**Comment by mkumatag** [comment]
> bug in alpha feature that did not get caught in our CI jobs. we need this in 1.26 itself!
> 
> /milestone v1.26

well, we didn't enable any alpha feature gates.. this was just happening in our regular CI test on ppc64le platform with single NUMA node(with id 1)

**Comment by klueska** [comment]
@PiotrProkop Thanks for the update.

Now that the `cadvisor` update has landed this is much cleaner and much more protected behind the feature gate.

/lgtm

**Comment by PiotrProkop** [comment]
Thanks, for the review.
/unhold

**Comment by PiotrProkop** [comment]
/retest
The error seems unrelated:
```
Nov  9 17:22:21.469: INFO: Waiting 10s to retry failed RunHostCmd: error running /home/prow/go/src/k8s.io/kubernetes/_output/bin/kubectl --server=https://127.0.0.1:35701/ --kubeconfig=/root/.kube/kind-test-config --namespace=statefulset-6261 exec ss-1 -- /bin/sh -x -c mv -v /tmp/index.html /usr/local/apache2/htdocs/ || true:
        Command stdout:
        stderr:
        Error from server (NotFound): pods "ss-1" not found
        error:
        exit status 1
```

**Pull request merged**